src/reactviews/pages/QueryResult/slickgrid.tsx

```diff
@@ -0,0 +1,182 @@
+						return hyperLinkFormatter(row, cell, value, columnDef, dataContext);
+					} else if (isJsonCell(value) && props.resultSetSummary) {
+                        //TODO use showJsonAsLink config
+						props.resultSetSummary.columnInfo[i].isJson = true;
+						return hyperLinkFormatter(row, cell, value, columnDef, dataContext);
+					} else {
+						return textFormatter(row, cell, value, columnDef, dataContext, (DBCellValue.isDBCellValue(value) && value.isNull) ? NULL_CELL_CSS_CLASS : undefined);
+					}
+				},
+				// width: this.state.columnSizes && this.state.columnSizes[i] ? this.state.columnSizes[i] : undefined
+			};
+		});
+        // let options = {
+        //     enableCellNavigation: true,
+        //     enableColumnReorder: false
+        // };
+
+        let div = document.createElement('div');
+        div.id = 'grid';
+        div.className = 'grid-panel';
+        div.style.display = 'inline-block';
+
+        //TODO: eventually need to calculate snapshot button width and subtract
+        // let actionBarWidth = this.showActionBar ? ACTIONBAR_WIDTH : 0;
+        // this.tableContainer.style.width = `calc(100% - ${actionBarWidth}px)`;
+
+
+
+        let tableOptions: Slick.GridOptions<Slick.SlickData> = {
+            rowHeight: ROW_HEIGHT,
+            showRowNumber: true,
+            forceFitColumns: false,
+            defaultColumnWidth: 120
+        };
+        let rowNumberColumn = new RowNumberColumn<Slick.SlickData>({ autoCellSelection: false });
+        columns.unshift(rowNumberColumn.getColumnDefinition());
+
+        let collection = new VirtualizedCollection<any>(
+            50,
+            _index => { },
+            props.resultSetSummary?.rowCount ?? 0,
+            props.loadFunc
+        );
+
+        let dataProvider = new HybridDataProvider(collection,
+            (_startIndex, _count) => Promise.resolve([]),
+            (data: DbCellValue) => {
+                if (!data || data.isNull) {
+                    return undefined;
+                }
+                // If the string only contains whitespaces, it will be treated as empty string to make the filtering easier.
+                // Note: this is the display string and does not impact the export/copy features.
+                return data.displayValue.trim() === '' ? '' : data.displayValue;
+            },
+            {
+                inMemoryDataProcessing: true,
+            },
+            undefined,
+            undefined);
+        var table = new Table(div, defaultTableStyles, { dataProvider: dataProvider, columns: columns }, tableOptions);
+
+        collection.setCollectionChangedCallback((startIndex, count) => {
+            let refreshedRows = range(startIndex, startIndex + count);
+            table.invalidateRows(refreshedRows, true);
+        });
+        table.updateRowCount();
+
+        let grid = document.body.appendChild(div);
+        const elm = document.getElementById('grid')!;
+        document.body.removeChild(grid);
```

Comment:
I see that we are clearing out the old grid. What about old Table object? Is that getting disposed properly?